### ISSUE TITLE
Refactor clonotype_imbalance

### ISSUE THREAD

**Issue by grst** [icbi-lab/scirpy#160]
Analogous to #156, at some point

- [ ] make volcano and dotplot a different function
- [ ] make clear what is computed by the tool and what on-the-fly
- [ ] more compact representation for dotplot, potentially a heatmap could work.

**Comment by grst** [comment]
Potentially, `clonotype_imbalance` will be made obsolte by `clonotype_modularity`. 
For now, I marked `clonotype_imbalance` as deprecated with a note asking the user to switch to clonotype modularity, or drop an issue if that doesn't fit their use-case. 

Let's see how that goes, in the meanwhile, `clonotype_imbalance` will remain as is.

**Comment by swstkm** [comment]
Hello, I would like to work on this. Is anyone already on it?

The deprecation was reverted after #330 and I don't see an open PR touching it,
so I assume it's still current.

Would you prefer the whole list, or a minimal first step? The Fisher's exact test bit looks self-contained.

**Comment by grst** [comment]
Yes, this is still a thing. If you want to work on this, as a first step, it would be good to come up with a proposal what the API of the new functions that replace `clonotype_imbalance` would look like.
